src/modules/fw_wind_estimator/FixedwingWindEstimator.cpp

```diff
@@ -0,0 +1,307 @@
+		// get rotation between NED frames
+		_local_velocity = Vector3f(vehicle_local_position.vx, vehicle_local_position.vy, vehicle_local_position.vz);
+	}
+}
+
+void
+FixedwingWindEstimator::vehicle_acceleration_poll()
+{
+	//vehicle_local_position_s pos;
+	///TODO: We should probably get it from the imu, not the local one?
+	vehicle_acceleration_s vehicle_acceleration;
+
+	if (_vehicle_acceleration_sub.update(&vehicle_acceleration)) {
+		Dcmf R_ib(_attitude);
+		_acceleration = R_ib * Vector3f(vehicle_acceleration.xyz);
+	}
+}
+
+matrix::Vector3f FixedwingWindEstimator::compute_wind_estimate()
+{
+	float _rho{1.225};
+
+	Dcmf R_ib(_attitude);
+	Dcmf R_bi(R_ib.transpose());
+	// compute expected AoA from g-forces:
+	matrix::Vector3f body_force = _mass * R_bi * (_acceleration + _gravity);
+
+	// ***************** NEW COMPUTATION FROM MATLAB CALIBRATION **********************
+	float speed = fmaxf(_calibrated_airspeed, _stall_airspeed);
+	float u_approx = _true_airspeed;
+	float v_approx = body_force(1) * _true_airspeed / (0.5f * _rho * powf(speed, 2) * _wing_area * _C_B1);
+	float w_approx = (-body_force(2) * _true_airspeed / (0.5f * _rho * powf(speed, 2) * _wing_area) - _C_A0) / _C_A1;
+	matrix::Vector3f vel_air = matrix::Vector3f{u_approx, v_approx, w_approx};
+	return vel_air;
+}
+
+void FixedwingWindEstimator::Run()
+{
+	if (should_exit()) {
+		_vehicle_angular_velocity_sub.unregisterCallback();
+		exit_and_cleanup();
+		return;
+	}
+
+	perf_begin(_loop_perf);
+
+	// only run controller if angular velocity changed
+	if (_vehicle_angular_velocity_sub.updated() || (hrt_elapsed_time(&_last_run) > 20_ms)) { //TODO rate!
+
+		// only update parameters if they changed
+		bool params_updated = _parameter_update_sub.updated();
+
+		// check for parameter updates
+		if (params_updated) {
+			// clear update
+			parameter_update_s pupdate;
+			_parameter_update_sub.copy(&pupdate);
+
+			// update parameters from storage
+			updateParams();
+			parameters_update();
+		}
+
+		float dt = 0.f;
+
+		static constexpr float DT_MIN = 0.002f;
+		static constexpr float DT_MAX = 0.04f;
+
+		vehicle_angular_velocity_s vehicle_angular_velocity{};
+
+		if (_vehicle_angular_velocity_sub.copy(&vehicle_angular_velocity)) {
+			dt = math::constrain((vehicle_angular_velocity.timestamp_sample - _last_run) * 1e-6f, DT_MIN, DT_MAX);
+			_last_run = vehicle_angular_velocity.timestamp_sample;
+		}
+
+		if (dt < DT_MIN || dt > DT_MAX) {
+			const hrt_abstime time_now_us = hrt_absolute_time();
+			dt = math::constrain((time_now_us - _last_run) * 1e-6f, DT_MIN, DT_MAX);
+			_last_run = time_now_us;
+		}
+
+		_vehicle_control_mode_sub.update(&_vcontrol_mode);
+
+		vehicle_land_detected_poll();
+		airspeed_poll();
+		vehicle_attitude_poll();
+		vehicle_local_position_poll();
+
+		// Do not compute wind estimate under stall speed
+		if (_calibrated_airspeed > _stall_airspeed) {
+			matrix::Vector3f air_velocity_body = compute_wind_estimate();
+
+			Dcmf R_ib(_attitude);
```

Comment:
as for the other ones, only convert to `Dcmf` when you need to rotate several vectors